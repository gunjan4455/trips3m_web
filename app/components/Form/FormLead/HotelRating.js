import React from 'react';
import PropTypes from 'prop-types';

const hotelPrefCustomizations = {
  star2: 32,
  star3: 8,
  star4: 7,
  star5: 6,
  noHotel: 96,
  customizationId: 2,
  noHotelCustomizationId: 22
};

class HotelRating extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      hotels: {
        star2: false,
        star3: false,
        star4: false,
        star5: false,
        noHotel: false
      },
    };
    this.handleChange = this.handleChange.bind(this);
    this.getHotelPrefs = this.getHotelPrefs.bind(this);
    this.checkInitialValues = this.checkInitialValues.bind(this);
    this.checkInitialValues(props.formValues);
  }

  getHotelPrefs() {
    let isHotelSelected = false;
    const retValue = Object.keys(this.state.hotels)
      .filter(hotelStar => this.state.hotels[hotelStar])
      .map((hotelStar) => {
        const customizationId = hotelPrefCustomizations.customizationId;
        if (!this.state.hotels.noHotel) {
          isHotelSelected = true;
          return {
            option_id: hotelPrefCustomizations[hotelStar].toString(),
            customization_id: customizationId.toString()
          };
        }
        return {};
      });
    if (isHotelSelected) {
      this.props.dispatchRxFormAction({
        option_id: '95',
        customization_id: hotelPrefCustomizations.noHotelCustomizationId.toString()
      }, 'hotel_needed_answer_attributes');
    } else if (this.state.hotels.noHotel) {
      this.props.dispatchRxFormAction('', 'budget');
      this.props.dispatchRxFormAction({
        option_id: '96',
        customization_id: hotelPrefCustomizations.noHotelCustomizationId.toString()
      }, 'hotel_needed_answer_attributes');
    } else {
      this.props.dispatchRxFormAction('', 'hotel_needed_answer_attributes');
    }
    if (retValue.length < 1) {
      return '';
    }
    return retValue;
  }

  checkInitialValues(formValues) {
    const hotelAttributes = formValues.hotel_category_answers_attributes || [];
    const answerAttributes = formValues.hotel_needed_answer_attributes || {};
    hotelAttributes.forEach((obj) => {
      if (obj.option_id === hotelPrefCustomizations.star2.toString()) {
        this.state = {
          ...this.state,
          hotels: {
            ...this.state.hotels,
            star2: true
          }
        };
      } else if (obj.option_id === hotelPrefCustomizations.star3.toString()) {
        this.state = {
          ...this.state,
          hotels: {
            ...this.state.hotels,
            star3: true
          }
        };
      } else if (obj.option_id === hotelPrefCustomizations.star4.toString()) {
        this.state = {
          ...this.state,
          hotels: {
            ...this.state.hotels,
            star4: true
          }
        };
      } else if (obj.option_id === hotelPrefCustomizations.star5.toString()) {
        this.state = {
          ...this.state,
          hotels: {
            ...this.state.hotels,
            star5: true
          }
        };
      }
    });
    if (answerAttributes.option_id === '96') {
      this.state = {
        ...this.state,
        hotels: {
          ...this.state.hotels,
          noHotel: true
        }
      };
    }
  }

  handleChange(event) {
    const hotelPrefs = {};
    if (event.target.checked) {
      if (event.target.name === 'noHotel') {
        hotelPrefs.star2 = false;
        hotelPrefs.star3 = false;
        hotelPrefs.star4 = false;
        hotelPrefs.star5 = false;
      } else {
        hotelPrefs.noHotel = false;
      }
    }
    this.setState({
      hotels: {
        ...this.state.hotels,
        ...hotelPrefs,
        [event.target.name]: event.target.checked
      }
    }, () => { this.props.input.onChange(this.getHotelPrefs()); });
  }

  render() {
    return (
      <div className="hotelratingDiv at_hotelratingdiv">
        <p className="fw9 iblock m0 pfc3">Preferred Hotel Category
          <span className="pfc4 fw4 f12"> (Rating)</span>
          <em className="pfc2 f16 fwb">*</em>
        </p>
        <input
          type="checkbox"
          name="noHotel"
          id="No-hotel"
          className="No-hotel-input hide"
          checked={this.state.hotels.noHotel}
          onChange={this.handleChange}
        />
        <label htmlFor="No-hotel"
               className="text-uppercase pfc4 iblock fright fw4 cursorP pt3 f12 at_nohotel mb0">No Hotel</label>
        <div className="row row- pfc3">
          <div className="col-xs-3 p0">
            <input
              type="checkbox"
              id="hotel-checkbox5"
              name="star5"
              className="checkbox-common"
              onChange={this.handleChange}
              checked={this.state.hotels.star5}
            />
            <label htmlFor="hotel-checkbox5" className="cursorP">5 Star</label>
          </div>
          <div className="col-xs-3 p0">
            <input
              type="checkbox"
              id="hotel-checkbox4"
              name="star4"
              className="checkbox-common"
              onChange={this.handleChange}
              checked={this.state.hotels.star4}
            />
            <label htmlFor="hotel-checkbox4" className="cursorP">4 Star</label>
          </div>
          <div className="col-xs-3 p0">
            <input
              type="checkbox"
              id="hotel-checkbox3"
              name="star3"
              className="checkbox-common"
              onChange={this.handleChange}
              checked={this.state.hotels.star3}
            />
            <label htmlFor="hotel-checkbox3" className="cursorP">3 Star</label>
          </div>
          <div className="col-xs-3 p0">
            <input
              type="checkbox"
              id="hotel-checkbox2"
              name="star2"
              className="checkbox-common"
              onChange={this.handleChange}
              checked={this.state.hotels.star2}
            />
            <label htmlFor="hotel-checkbox2" className="cursorP">2 Star</label>
          </div>
          {
            this.props.meta
              && this.props.meta.touched
              && (this.props.meta.error && <span className="input-error-text">{this.props.meta.error}</span>)
          }
        </div>
      </div>
    );
  }
}
HotelRating.propTypes = {
  input: PropTypes.object.isRequired,
  meta: PropTypes.object.isRequired,
  dispatchRxFormAction: PropTypes.func.isRequired,
  formValues: PropTypes.object
};

HotelRating.defaultProps = {
  formValues: {}
};

export default HotelRating;
