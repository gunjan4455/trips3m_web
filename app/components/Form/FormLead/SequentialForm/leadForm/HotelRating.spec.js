import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import HotelRating from "./HotelRating";

configure({ adapter: new Adapter() });

const mockDispatchRxFormAction = jest.fn();

const DEFAULT_PROPS = {
  input: {
    name: 'hotel_category_answers_attributes',
    value: [
      {
        option_id: '7',
        customization_id: '2'
      }
    ],
    onChange: () => {
    }
  },
  meta: {
    active: false,
    asyncValidating: false,
    autofilled: false,
    dirty: false,
    form: 'LeadForm',
    initial: [
      {
        option_id: '7',
        customization_id: '2'
      }
    ],
    invalid: false,
    pristine: true,
    submitting: false,
    submitFailed: false,
    touched: false,
    valid: true,
    visited: false
  },
  formValues: {
    email: null,
    dont_utilize_budget_new: true,
    budget_new: true,
    start_date_noidea: false,
    adult: '2',
    child: 0,
    isd_code: '+91',
    child_age_answers_attributes: [
      {
        user_input: '0',
        customization_id: '7',
        option_id: '29'
      },
      {
        user_input: '0',
        customization_id: '7',
        option_id: '30'
      },
      {
        user_input: '0',
        customization_id: '7',
        option_id: '31'
      }
    ],
    syncErrors: {
      email: 'Enter valid email id'
    },
    to_loc: 'Kerala',
    start_date_flexible: 'Sep\'19 - Week 1',
    start_date: '191-09-08',
    trip_days: '3',
    hotel_needed_answer_attributes: {
      option_id: '95',
      customization_id: '22'
    },
    hotel_category_answers_attributes: [
      {
        option_id: '7',
        customization_id: '2'
      }
    ]
  },
  isHotelSelected: true,
  dispatchRxFormAction: mockDispatchRxFormAction
};

const shallowWrapperHotelRating = (props) => {
  return shallow(<HotelRating {...props} />);
};

describe('lead form module', () => {

  let props = { ...DEFAULT_PROPS };

  afterEach(() => {
    props = { ...DEFAULT_PROPS };
  });

  it('should render correctly', () => {
    const wrapper = shallowWrapperHotelRating(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleChange properly', () => {
    const event = {
      target: {
        checked: true,
        name: 'noHotel'
      }
    };
    const expected_result = {
      noHotel: true,
      star2: false,
      star3: false,
      star4: false,
      star5: false,
    };

    const wrapper = shallowWrapperHotelRating(props);
    wrapper.instance().handleChange(event);
    expect(wrapper.state().hotels).toEqual(expected_result);
  });

  it('should call handleChange properly', () => {
    const event = {
      target: {
        checked: false,
        name: 'noHotel'
      }
    };
    const expected_result = {
      noHotel: false,
      star2: false,
      star3: false,
      star4: true,
      star5: false,
    };

    const wrapper = shallowWrapperHotelRating(props);
    wrapper.instance().handleChange(event);
    expect(wrapper.state().hotels).toEqual(expected_result);
  });

  it('should call handleChange properly', () => {
    const event = {
      target: {
        checked: false,
        name: 'Hotel'
      }
    };
    const expected_result = {
      noHotel: false,
      Hotel:false,
      star2: false,
      star3: false,
      star4: true,
      star5: false,
    };

    const wrapper = shallowWrapperHotelRating(props);
    wrapper.instance().handleChange(event);
    expect(wrapper.state().hotels).toEqual(expected_result);
  });

});
