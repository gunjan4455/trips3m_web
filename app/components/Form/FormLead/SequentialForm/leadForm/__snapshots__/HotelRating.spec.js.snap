// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lead form module should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HotelRating
    dispatchRxFormAction={[MockFunction]}
    formValues={
      Object {
        "adult": "2",
        "budget_new": true,
        "child": 0,
        "child_age_answers_attributes": Array [
          Object {
            "customization_id": "7",
            "option_id": "29",
            "user_input": "0",
          },
          Object {
            "customization_id": "7",
            "option_id": "30",
            "user_input": "0",
          },
          Object {
            "customization_id": "7",
            "option_id": "31",
            "user_input": "0",
          },
        ],
        "dont_utilize_budget_new": true,
        "email": null,
        "hotel_category_answers_attributes": Array [
          Object {
            "customization_id": "2",
            "option_id": "7",
          },
        ],
        "hotel_needed_answer_attributes": Object {
          "customization_id": "22",
          "option_id": "95",
        },
        "isd_code": "+91",
        "start_date": "191-09-08",
        "start_date_flexible": "Sep'19 - Week 1",
        "start_date_noidea": false,
        "syncErrors": Object {
          "email": "Enter valid email id",
        },
        "to_loc": "Kerala",
        "trip_days": "3",
      }
    }
    input={
      Object {
        "name": "hotel_category_answers_attributes",
        "onChange": [Function],
        "value": Array [
          Object {
            "customization_id": "2",
            "option_id": "7",
          },
        ],
      }
    }
    isHotelSelected={true}
    meta={
      Object {
        "active": false,
        "asyncValidating": false,
        "autofilled": false,
        "dirty": false,
        "form": "LeadForm",
        "initial": Array [
          Object {
            "customization_id": "2",
            "option_id": "7",
          },
        ],
        "invalid": false,
        "pristine": true,
        "submitFailed": false,
        "submitting": false,
        "touched": false,
        "valid": true,
        "visited": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="fw9 iblock m0 pfc3 mb8 f14"
        >
          Select Hotel Category
          <em
            className="pfc2 f16 fwb"
          >
            *
          </em>
        </p>,
        <div
          className="month-container flex flexWrap spaceBetween"
        >
          <div
            className="month-option "
          >
            <input
              checked={false}
              className="checkbox-common"
              id="hotel-checkbox5"
              name="star5"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="cursorP p0"
              htmlFor="hotel-checkbox5"
            >
              5 Star
            </label>
          </div>
          <div
            className="month-option active"
          >
            <input
              checked={true}
              className="checkbox-common"
              id="hotel-checkbox4"
              name="star4"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="cursorP p0"
              htmlFor="hotel-checkbox4"
            >
              4 Star
            </label>
          </div>
          <div
            className="month-option "
          >
            <input
              checked={false}
              className="checkbox-common"
              id="hotel-checkbox3"
              name="star3"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="cursorP p0"
              htmlFor="hotel-checkbox3"
            >
              3 Star
            </label>
          </div>
          <div
            className="month-option "
          >
            <input
              checked={false}
              className="checkbox-common"
              id="hotel-checkbox2"
              name="star2"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="cursorP p0"
              htmlFor="hotel-checkbox2"
            >
              2 Star
            </label>
          </div>
          <div
            className="month-option "
          >
            <input
              checked={false}
              className="checkbox-common"
              id="No-hotel"
              name="noHotel"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="cursorP p0"
              htmlFor="No-hotel"
            >
              Hotel Not Required
            </label>
          </div>
        </div>,
      ],
      "className": "hotelratingDiv at_hotelratingdiv mb15",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Select Hotel Category",
            <em
              className="pfc2 f16 fwb"
            >
              *
            </em>,
          ],
          "className": "fw9 iblock m0 pfc3 mb8 f14",
        },
        "ref": null,
        "rendered": Array [
          "Select Hotel Category",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "*",
              "className": "pfc2 f16 fwb",
            },
            "ref": null,
            "rendered": "*",
            "type": "em",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="month-option "
            >
              <input
                checked={false}
                className="checkbox-common"
                id="hotel-checkbox5"
                name="star5"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="cursorP p0"
                htmlFor="hotel-checkbox5"
              >
                5 Star
              </label>
            </div>,
            <div
              className="month-option active"
            >
              <input
                checked={true}
                className="checkbox-common"
                id="hotel-checkbox4"
                name="star4"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="cursorP p0"
                htmlFor="hotel-checkbox4"
              >
                4 Star
              </label>
            </div>,
            <div
              className="month-option "
            >
              <input
                checked={false}
                className="checkbox-common"
                id="hotel-checkbox3"
                name="star3"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="cursorP p0"
                htmlFor="hotel-checkbox3"
              >
                3 Star
              </label>
            </div>,
            <div
              className="month-option "
            >
              <input
                checked={false}
                className="checkbox-common"
                id="hotel-checkbox2"
                name="star2"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="cursorP p0"
                htmlFor="hotel-checkbox2"
              >
                2 Star
              </label>
            </div>,
            <div
              className="month-option "
            >
              <input
                checked={false}
                className="checkbox-common"
                id="No-hotel"
                name="noHotel"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="cursorP p0"
                htmlFor="No-hotel"
              >
                Hotel Not Required
              </label>
            </div>,
            false,
          ],
          "className": "month-container flex flexWrap spaceBetween",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  checked={false}
                  className="checkbox-common"
                  id="hotel-checkbox5"
                  name="star5"
                  onChange={[Function]}
                  type="checkbox"
                />,
                <label
                  className="cursorP p0"
                  htmlFor="hotel-checkbox5"
                >
                  5 Star
                </label>,
              ],
              "className": "month-option ",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "checked": false,
                  "className": "checkbox-common",
                  "id": "hotel-checkbox5",
                  "name": "star5",
                  "onChange": [Function],
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "5 Star",
                  "className": "cursorP p0",
                  "htmlFor": "hotel-checkbox5",
                },
                "ref": null,
                "rendered": "5 Star",
                "type": "label",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  checked={true}
                  className="checkbox-common"
                  id="hotel-checkbox4"
                  name="star4"
                  onChange={[Function]}
                  type="checkbox"
                />,
                <label
                  className="cursorP p0"
                  htmlFor="hotel-checkbox4"
                >
                  4 Star
                </label>,
              ],
              "className": "month-option active",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "checked": true,
                  "className": "checkbox-common",
                  "id": "hotel-checkbox4",
                  "name": "star4",
                  "onChange": [Function],
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "4 Star",
                  "className": "cursorP p0",
                  "htmlFor": "hotel-checkbox4",
                },
                "ref": null,
                "rendered": "4 Star",
                "type": "label",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  checked={false}
                  className="checkbox-common"
                  id="hotel-checkbox3"
                  name="star3"
                  onChange={[Function]}
                  type="checkbox"
                />,
                <label
                  className="cursorP p0"
                  htmlFor="hotel-checkbox3"
                >
                  3 Star
                </label>,
              ],
              "className": "month-option ",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "checked": false,
                  "className": "checkbox-common",
                  "id": "hotel-checkbox3",
                  "name": "star3",
                  "onChange": [Function],
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "3 Star",
                  "className": "cursorP p0",
                  "htmlFor": "hotel-checkbox3",
                },
                "ref": null,
                "rendered": "3 Star",
                "type": "label",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  checked={false}
                  className="checkbox-common"
                  id="hotel-checkbox2"
                  name="star2"
                  onChange={[Function]}
                  type="checkbox"
                />,
                <label
                  className="cursorP p0"
                  htmlFor="hotel-checkbox2"
                >
                  2 Star
                </label>,
              ],
              "className": "month-option ",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "checked": false,
                  "className": "checkbox-common",
                  "id": "hotel-checkbox2",
                  "name": "star2",
                  "onChange": [Function],
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "2 Star",
                  "className": "cursorP p0",
                  "htmlFor": "hotel-checkbox2",
                },
                "ref": null,
                "rendered": "2 Star",
                "type": "label",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  checked={false}
                  className="checkbox-common"
                  id="No-hotel"
                  name="noHotel"
                  onChange={[Function]}
                  type="checkbox"
                />,
                <label
                  className="cursorP p0"
                  htmlFor="No-hotel"
                >
                  Hotel Not Required
                </label>,
              ],
              "className": "month-option ",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "checked": false,
                  "className": "checkbox-common",
                  "id": "No-hotel",
                  "name": "noHotel",
                  "onChange": [Function],
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Hotel Not Required",
                  "className": "cursorP p0",
                  "htmlFor": "No-hotel",
                },
                "ref": null,
                "rendered": "Hotel Not Required",
                "type": "label",
              },
            ],
            "type": "div",
          },
          false,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="fw9 iblock m0 pfc3 mb8 f14"
          >
            Select Hotel Category
            <em
              className="pfc2 f16 fwb"
            >
              *
            </em>
          </p>,
          <div
            className="month-container flex flexWrap spaceBetween"
          >
            <div
              className="month-option "
            >
              <input
                checked={false}
                className="checkbox-common"
                id="hotel-checkbox5"
                name="star5"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="cursorP p0"
                htmlFor="hotel-checkbox5"
              >
                5 Star
              </label>
            </div>
            <div
              className="month-option active"
            >
              <input
                checked={true}
                className="checkbox-common"
                id="hotel-checkbox4"
                name="star4"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="cursorP p0"
                htmlFor="hotel-checkbox4"
              >
                4 Star
              </label>
            </div>
            <div
              className="month-option "
            >
              <input
                checked={false}
                className="checkbox-common"
                id="hotel-checkbox3"
                name="star3"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="cursorP p0"
                htmlFor="hotel-checkbox3"
              >
                3 Star
              </label>
            </div>
            <div
              className="month-option "
            >
              <input
                checked={false}
                className="checkbox-common"
                id="hotel-checkbox2"
                name="star2"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="cursorP p0"
                htmlFor="hotel-checkbox2"
              >
                2 Star
              </label>
            </div>
            <div
              className="month-option "
            >
              <input
                checked={false}
                className="checkbox-common"
                id="No-hotel"
                name="noHotel"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="cursorP p0"
                htmlFor="No-hotel"
              >
                Hotel Not Required
              </label>
            </div>
          </div>,
        ],
        "className": "hotelratingDiv at_hotelratingdiv mb15",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Select Hotel Category",
              <em
                className="pfc2 f16 fwb"
              >
                *
              </em>,
            ],
            "className": "fw9 iblock m0 pfc3 mb8 f14",
          },
          "ref": null,
          "rendered": Array [
            "Select Hotel Category",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "*",
                "className": "pfc2 f16 fwb",
              },
              "ref": null,
              "rendered": "*",
              "type": "em",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="month-option "
              >
                <input
                  checked={false}
                  className="checkbox-common"
                  id="hotel-checkbox5"
                  name="star5"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="cursorP p0"
                  htmlFor="hotel-checkbox5"
                >
                  5 Star
                </label>
              </div>,
              <div
                className="month-option active"
              >
                <input
                  checked={true}
                  className="checkbox-common"
                  id="hotel-checkbox4"
                  name="star4"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="cursorP p0"
                  htmlFor="hotel-checkbox4"
                >
                  4 Star
                </label>
              </div>,
              <div
                className="month-option "
              >
                <input
                  checked={false}
                  className="checkbox-common"
                  id="hotel-checkbox3"
                  name="star3"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="cursorP p0"
                  htmlFor="hotel-checkbox3"
                >
                  3 Star
                </label>
              </div>,
              <div
                className="month-option "
              >
                <input
                  checked={false}
                  className="checkbox-common"
                  id="hotel-checkbox2"
                  name="star2"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="cursorP p0"
                  htmlFor="hotel-checkbox2"
                >
                  2 Star
                </label>
              </div>,
              <div
                className="month-option "
              >
                <input
                  checked={false}
                  className="checkbox-common"
                  id="No-hotel"
                  name="noHotel"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="cursorP p0"
                  htmlFor="No-hotel"
                >
                  Hotel Not Required
                </label>
              </div>,
              false,
            ],
            "className": "month-container flex flexWrap spaceBetween",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    checked={false}
                    className="checkbox-common"
                    id="hotel-checkbox5"
                    name="star5"
                    onChange={[Function]}
                    type="checkbox"
                  />,
                  <label
                    className="cursorP p0"
                    htmlFor="hotel-checkbox5"
                  >
                    5 Star
                  </label>,
                ],
                "className": "month-option ",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "checked": false,
                    "className": "checkbox-common",
                    "id": "hotel-checkbox5",
                    "name": "star5",
                    "onChange": [Function],
                    "type": "checkbox",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "5 Star",
                    "className": "cursorP p0",
                    "htmlFor": "hotel-checkbox5",
                  },
                  "ref": null,
                  "rendered": "5 Star",
                  "type": "label",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    checked={true}
                    className="checkbox-common"
                    id="hotel-checkbox4"
                    name="star4"
                    onChange={[Function]}
                    type="checkbox"
                  />,
                  <label
                    className="cursorP p0"
                    htmlFor="hotel-checkbox4"
                  >
                    4 Star
                  </label>,
                ],
                "className": "month-option active",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "checked": true,
                    "className": "checkbox-common",
                    "id": "hotel-checkbox4",
                    "name": "star4",
                    "onChange": [Function],
                    "type": "checkbox",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "4 Star",
                    "className": "cursorP p0",
                    "htmlFor": "hotel-checkbox4",
                  },
                  "ref": null,
                  "rendered": "4 Star",
                  "type": "label",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    checked={false}
                    className="checkbox-common"
                    id="hotel-checkbox3"
                    name="star3"
                    onChange={[Function]}
                    type="checkbox"
                  />,
                  <label
                    className="cursorP p0"
                    htmlFor="hotel-checkbox3"
                  >
                    3 Star
                  </label>,
                ],
                "className": "month-option ",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "checked": false,
                    "className": "checkbox-common",
                    "id": "hotel-checkbox3",
                    "name": "star3",
                    "onChange": [Function],
                    "type": "checkbox",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "3 Star",
                    "className": "cursorP p0",
                    "htmlFor": "hotel-checkbox3",
                  },
                  "ref": null,
                  "rendered": "3 Star",
                  "type": "label",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    checked={false}
                    className="checkbox-common"
                    id="hotel-checkbox2"
                    name="star2"
                    onChange={[Function]}
                    type="checkbox"
                  />,
                  <label
                    className="cursorP p0"
                    htmlFor="hotel-checkbox2"
                  >
                    2 Star
                  </label>,
                ],
                "className": "month-option ",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "checked": false,
                    "className": "checkbox-common",
                    "id": "hotel-checkbox2",
                    "name": "star2",
                    "onChange": [Function],
                    "type": "checkbox",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "2 Star",
                    "className": "cursorP p0",
                    "htmlFor": "hotel-checkbox2",
                  },
                  "ref": null,
                  "rendered": "2 Star",
                  "type": "label",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    checked={false}
                    className="checkbox-common"
                    id="No-hotel"
                    name="noHotel"
                    onChange={[Function]}
                    type="checkbox"
                  />,
                  <label
                    className="cursorP p0"
                    htmlFor="No-hotel"
                  >
                    Hotel Not Required
                  </label>,
                ],
                "className": "month-option ",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "checked": false,
                    "className": "checkbox-common",
                    "id": "No-hotel",
                    "name": "noHotel",
                    "onChange": [Function],
                    "type": "checkbox",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Hotel Not Required",
                    "className": "cursorP p0",
                    "htmlFor": "No-hotel",
                  },
                  "ref": null,
                  "rendered": "Hotel Not Required",
                  "type": "label",
                },
              ],
              "type": "div",
            },
            false,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
